core/src/main/java/ai/timefold/solver/core/preview/api/move/MutableSolutionView.java

```diff
@@ -119,14 +119,41 @@ <Entity_, Value_> void changeVariable(PlanningVariableMetaModel<Solution_, Entit
      * Moves a value within one entity's {@link PlanningListVariable planning list variable}.
      *
      * @param variableMetaModel Describes the variable to be changed.
-     * @param entity The entity whose variable value is to be changed.
+     * @param sourceEntity The entity whose variable value is to be changed.
      * @param sourceIndex >= 0
      * @param destinationIndex >= 0
      * @return the value that was moved; null if nothing was moved
      * @throws IndexOutOfBoundsException if the index is out of bounds
      */
     <Entity_, Value_> @Nullable Value_ moveValueInList(
-            PlanningListVariableMetaModel<Solution_, Entity_, Value_> variableMetaModel, Entity_ entity, int sourceIndex,
+            PlanningListVariableMetaModel<Solution_, Entity_, Value_> variableMetaModel, Entity_ sourceEntity, int sourceIndex,
             int destinationIndex);
 
+    /**
+     * Swaps two values between two entities' {@link PlanningListVariable planning list variable}.
+     *
+     * @param variableMetaModel Describes the variable to be changed.
+     * @param leftEntity The first entity whose variable value is to be swapped.
+     * @param leftIndex >= 0
+     * @param rightEntity The second entity whose variable value is to be swapped.
+     * @param rightIndex >= 0
+     * @throws IndexOutOfBoundsException if the index is out of bounds
+     */
+    <Entity_, Value_> void swapValuesBetweenLists(
+            PlanningListVariableMetaModel<Solution_, Entity_, Value_> variableMetaModel, Entity_ leftEntity, int leftIndex,
+            Entity_ rightEntity, int rightIndex);
+
+    /**
+     * Swaps two values within one entity's {@link PlanningListVariable planning list variable}.
+     *
+     * @param variableMetaModel Describes the variable to be changed.
+     * @param entity The entity whose variable values are to be swapped.
+     * @param leftIndex >= 0
+     * @param rightIndex >= 0
+     * @throws IndexOutOfBoundsException if the index is out of bounds
+     */
+    <Entity_, Value_> void swapValuesInList(
```

Comment:
I have not checked the implementation logic yet, but I wonder why we don't use a default implementation that calls the previous method. This way, we can delegate the responsibility of identifying whether it is the same entity or not to the previous method.

Reply:
There are two ways of looking at this - one from the user's point of view, and the other from ours. In this case, I choose the user's.

In their move code, they want to have the least possible logic. And therefore if they really want to swap values in the same list, they want to do:

     swap(list, x, y)

And not:

     swap(list, x, list, y)

Reply:
I agree that we should have both methods, but it seems we could also have just one implementation:

```
    default <Entity_, Value_> @Nullable Value_ moveValueInList(
            PlanningListVariableMetaModel<Solution_, Entity_, Value_> variableMetaModel, Entity_ entity, int sourceIndex,
            int destinationIndex) {
        return moveValueBetweenLists(variableMetaModel, entity, sourceIndex, entity, destinationIndex);
    }
```

`moveValueBetweenLists` would be responsible for determining whether it is the same entity or not.